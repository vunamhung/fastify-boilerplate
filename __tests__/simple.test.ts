import * as fastify from "fastify";
import { Server, IncomingMessage, ServerResponse } from "http";

import statusRoutes from "../src/modules/routes/status";

describe("/status", () => {
  let server: fastify.FastifyInstance<Server, IncomingMessage, ServerResponse>;

  beforeAll(() => {});

  beforeEach(async () => {
    server = fastify({});
    // eslint-disable-next-line global-require
    server.register(statusRoutes);
    await server.ready();

    jest.clearAllMocks();
  });

  it("GET returns 200", async (done) => {
    const response = await server.inject({ method: "GET", url: "/status" });
    expect(response.statusCode).toEqual(200);
    const payload: { date: Date; works: boolean } = JSON.parse(
      response.payload
    );
    expect(payload).toMatchSnapshot({ date: expect.any(String), works: true });

    done();
  });

  it("POST returns 404", async (done) => {
    const response = await server.inject({ method: "POST", url: "/status" });
    expect(response.statusCode).toEqual(404);
    expect(response.payload).toMatchSnapshot();

    done();
  });
});
