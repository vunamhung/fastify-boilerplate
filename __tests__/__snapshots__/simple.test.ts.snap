// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/status GET returns 200 1`] = `
Object {
  "date": Any<String>,
  "works": true,
}
`;

exports[`/status POST returns 404 1`] = `"{\\"message\\":\\"Route POST:/status not found\\",\\"error\\":\\"Not Found\\",\\"statusCode\\":404}"`;
